### PR TITLE
Remove transcripts from indexes

### PR DESCRIPTION
**Pull request by masaball**
Related issue: #5892

### CODE REVIEW

File: app/controllers/supplemental_files_controller.rb [hunk 1]
```diff
@@ -110,6 +110,7 @@ def update
     @supplemental_file.attach_file(attachment) if attachment
 
     raise Avalon::SaveError, @supplemental_file.errors.full_messages unless @supplemental_file.save
+    raise Avalon::SaveError, @object.errors[:supplemental_files_json] unless @object.save
```
Comment (cjcolvar) [inline app/controllers/supplemental_files_controller.rb:113]: If the expected change is to the parent object's solr document then maybe it makes more sense to have the SupplementalFile model have an after_save callback which calls `update_index` on the parent object.  But maybe that would put too strong of a link between the two models (I think callbacks between models like this have caused us problems in the past and has the potential for unintentionally causing lots of saves/index updates) and doing the update here in the controller makes more sense.  If that's the case then I think the parent object doesn't actually need to be saved but only have `update_index` called on it.
```suggestion
    # Updates parent object's solr document
    @object.update_index
```

**Review by cjcolvar — COMMENTED** [review]
Looks really good!  I had one question/suggestion around updates to a supplemental file but otherwise seems great!

File: app/controllers/supplemental_files_controller.rb (hunk 1 above)
Comment (masaball) [reply app/controllers/supplemental_files_controller.rb:113]: Yeah, I hesitate to use an after save callback. I started to try that when I was first working on this but it was causing a lot of issues with the testing suite and I imagine the finickiness could potentially translate to the live environs.

I will try `update_index` and see how that goes. I missed that that was a method that exists but is exactly what I was looking for.

**Review by cjcolvar — APPROVED** [review]
Nice!  The failed test is an unrelated flaky test so feel free to go ahead and merge.

**Pull request merged**